library/tiqr/Tiqr/DeviceStorage/Abstract.php

```diff
@@ -39,26 +39,27 @@ abstract class Tiqr_DeviceStorage_Abstract
     /**
      * get a deviceToken for a certain notificationToken.
      * @param String $notificationToken
-     * @return String deviceToken
+     * @return String|bool deviceToken, false on error
      */
-    public abstract function getDeviceToken($notificationToken);
+    public abstract function getDeviceToken(string $notificationToken);
```

Comment:
Consider making the return type of this method stable. So return string on the positive flow. And throw an exception on error.

Reply:
Good catch. The DeviceStorage did not get the basic cleanup.